### ISSUE TITLE
V7- Error 'No Document Exists with id 'xxx' when saving a Content Template Blueprint

### ISSUE THREAD

**Issue by marcemarc** [umbraco/Umbraco-CMS#8499]
## Umbraco version

7.15.5

### Bug summary

In Umbraco 7.15.5, when creating a new Content Template from an existing content item, and when updating that content item from the Settings section - the Content Template IS saved, however an error message is thrown:

An error occured
No Document exists with id '1147'

(where 1147 is the id of the Content Template that has been created, or is being updated)

![image](https://user-images.githubusercontent.com/260802/88663647-d648e600-d0d3-11ea-9b80-42ad0735732e.png)

This is the stacktrace:

```
Exception Details
System.ArgumentException: No Document exists with id '1147'
Stacktrace
at umbraco.cms.businesslogic.web.Document.setupNode() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\umbraco.cms\businesslogic\web\Document.cs:line 1410
   at umbraco.cms.businesslogic.CMSNode..ctor(Int32 Id) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\umbraco.cms\businesslogic\CMSNode.cs:line 394
   at umbraco.cms.businesslogic.Content..ctor(Int32 id) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\umbraco.cms\businesslogic\Content.cs:line 57
   at umbraco.cms.businesslogic.web.Document..ctor(Int32 id) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\umbraco.cms\businesslogic\web\Document.cs:line 68
   at Umbraco.Web.Cache.UnpublishedPageCacheRefresher.Refresh(String jsonPayload) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Cache\UnpublishedPageCacheRefresher.cs:line 169
   at Umbraco.Core.Sync.ServerMessengerBase.DeliverLocal(ICacheRefresher refresher, MessageType messageType, IEnumerable`1 ids, String json) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Sync\ServerMessengerBase.cs:line 211
   at Umbraco.Core.Sync.ServerMessengerBase.Deliver(IEnumerable`1 servers, ICacheRefresher refresher, MessageType messageType, IEnumerable`1 ids, String json) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Sync\ServerMessengerBase.cs:line 323
   at Umbraco.Core.Sync.ServerMessengerBase.PerformRefresh(IEnumerable`1 servers, ICacheRefresher refresher, String jsonPayload) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Sync\ServerMessengerBase.cs:line 74
   at Umbraco.Web.Cache.DistributedCache.RefreshByJson(Guid factoryGuid, String jsonPayload) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Cache\DistributedCache.cs:line 173
   at Umbraco.Web.Cache.DistributedCacheExtensions.RefreshUnpublishedPageCache(DistributedCache dc, IContent[] content) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Cache\DistributedCacheExtensions.cs:line 175
   at Umbraco.Web.Cache.CacheRefresherEventHandler.ContentService_SavedBlueprint(IContentService sender, SaveEventArgs`1 e) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Cache\CacheRefresherEventHandler.cs:line 397
   at Umbraco.Core.Events.EventDefinition`2.RaiseEvent() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Events\EventDefinition.cs:line 69
   at Umbraco.Core.Events.ScopeEventDispatcher.ScopeExitCompleted() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Events\ScopeEventDispatcher.cs:line 24
   at Umbraco.Core.Events.ScopeEventDispatcherBase.ScopeExit(Boolean completed) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Events\ScopeEventDispatcherBase.cs:line 338
   at Umbraco.Core.Scoping.Scope.<>c__DisplayClass70_0.<RobustExit>b__1() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 456
   at Umbraco.Core.Scoping.Scope.TryFinally(Int32 index, Action[] actions) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 495
   at Umbraco.Core.Scoping.Scope.TryFinally(Int32 index, Action[] actions) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 499
   at Umbraco.Core.Scoping.Scope.TryFinally(Action[] actions) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 487
   at Umbraco.Core.Scoping.Scope.RobustExit(Boolean completed, Boolean onException) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 443
   at Umbraco.Core.Scoping.Scope.DisposeLastScope() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 423
   at Umbraco.Core.Scoping.Scope.Dispose() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Scoping\Scope.cs:line 386
   at Umbraco.Core.Persistence.UnitOfWork.ScopeUnitOfWork.DisposeResources() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Persistence\UnitOfWork\ScopeUnitOfWork.cs:line 257
   at Umbraco.Core.DisposableObjectSlim.Dispose(Boolean disposing) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\DisposableObjectSlim.cs:line 33
   at Umbraco.Core.DisposableObjectSlim.Dispose() in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\DisposableObjectSlim.cs:line 20
   at Umbraco.Core.Services.ContentService.SaveBlueprint(IContent content, Int32 userId) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Core\Services\ContentService.cs:line 1351
   at Umbraco.Web.Editors.ContentController.<>c__DisplayClass26_0.<PostSaveBlueprint>b__0(IContent content) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Editors\ContentController.cs:line 592
   at Umbraco.Web.Editors.ContentController.PostSaveInternal(ContentItemSave contentItem, Func`2 saveMethod) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Editors\ContentController.cs:line 675
   at Umbraco.Web.Editors.ContentController.PostSaveBlueprint(ContentItemSave contentItem) in G:\devops\Umbraco-CMS-release-7.15.5\Umbraco-CMS-release-7.15.5\src\Umbraco.Web\Editors\ContentController.cs:line 587
   at lambda_method(Closure , Object , Object[] )
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass6_2.<GetExecutor>b__2(Object instance, Object[] methodParameters)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext, IDictionary`2 arguments, CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__15.MoveNext()
```

Following the stack trace, leads us to the ContentService 'SavedBlueprint' event - which is action after the save of the item, which is why the updates work.

So a quick search reveals that the Umbraco Core is looking out for that particular event and handling it in Umbraco.Web/Cache/CacheRefresherEventHandler.cs

This is listening out for changes on business objects in order to alert other servers (when load balancing?) that data has changed, and they should invalidate their caches.

This is how the SavedBlueprint is handled:

```
        static void ContentService_SavedBlueprint(IContentService sender, SaveEventArgs<IContent> e)
        {
            DistributedCache.Instance.RefreshUnpublishedPageCache(e.SavedEntities.ToArray());
        }
```
So my first question here (as I don't really understand this :-)) is why are we 'refreshing unpublished page cache across servers' when a blueprint has changed? in a load balanced environment - you only have one server for editing content, and blueprints are only used in the backoffice? 

If I comment this line out, the error goes away! but not understanding what it's trying to achieve, feel this isn't necessarily the problem, unless it just doesn't need to do this, but it looks like it's been like this for a while, so why is it broken now in 7.15.5?

Following a bit further, the UnpublishedPageCacheRefresher.Refresh method by jsonPayload is eventually called:

```
 public void Refresh(string jsonPayload)
        {
            ClearAllIsolatedCacheByEntityType<PublicAccessEntry>();

            foreach (var payload in DeserializeFromJsonPayload(jsonPayload))
            {
                ClearRepositoryCacheItemById(payload.Id);
                ClearRepositoryCacheItemById(payload.Key);
                ClearAllIsolatedCacheByEntityType<PublicAccessEntry>();

                if (payload.Operation == OperationType.Deleted)
                {
                    ApplicationContext.Current.Services.IdkMap.ClearCache(payload.Id);
                    content.Instance.ClearPreviewXmlContent(payload.Id);
                    base.Remove(payload.Id);
                }

                if (payload.Operation == OperationType.Refresh)
                {
                    content.Instance.UpdateSortOrder(payload.Id);
                    var d = new Document(payload.Id);
                    content.Instance.UpdateDocumentCache(d);
                    content.Instance.UpdatePreviewXmlContent(d);

                    base.Refresh(payload.Id);
                }
            }

            DistributedCache.Instance.ClearDomainCacheOnCurrentServer();

            OnCacheUpdated(Instance, new CacheRefresherEventArgs(jsonPayload, MessageType.RefreshByJson));
        }
```
And you can see this line:

          `          var d = new Document(payload.Id);`

Where it tries to get the 'Document' by id is where the error is eventually thrown in the rather ancient: setupNode() method:

```
[Obsolete("Obsolete", false)]
        protected override void setupNode()
        {
            if (Id == -1 || Id == -20)
            {
                base.setupNode();
                return;
            }

            var content = Version == Guid.Empty
                           ? ApplicationContext.Current.Services.ContentService.GetById(Id)
                           : ApplicationContext.Current.Services.ContentService.GetByVersion(Version);

            if(content == null)
                throw new ArgumentException(string.Format("No Document exists with id '{0}'", Id));

            SetupNode(content);
        }
```
Here the GetByVersion method is executed:

```
var sql = GetBaseQuery(BaseQueryType.FullSingle);
            //TODO: cmsContentVersion.VersionId has a Unique Index constraint applied, seems silly then to also add OrderByDescending since it would be impossible to return more than one.
            sql.Where("cmsContentVersion.VersionId = @VersionId", new { VersionId = versionId });
            sql.OrderByDescending<ContentVersionDto>(x => x.VersionDate, SqlSyntax);

            var dto = Database.Fetch<DocumentDto, ContentVersionDto, ContentDto, NodeDto, DocumentPublishedReadOnlyDto>(sql).FirstOrDefault();
```

and no object is retrieved? I'm guessing either the guid is wrong for the blueprint - or blueprints aren't stored in the cmsContentVersion database table?

Anyway this is why the error is thrown.

It looks like the new Document() code was added here in 7.15.0:
https://github.com/umbraco/Umbraco-CMS/commit/0676726bfcf2343e729af869420975ab9242701b

as part of 'refactoring preview' - could this be the point it became broken, I am unsure as I have a live 7.15.3 site where this functionality works without error... but it is a 7.15.5 upgrade on a different site where I have encountered the problem - and I've verified the issue with a fresh download of 7.15.5 from the release tag in the repo. 

Is there a scenario where the CacheRefresherEventHandler doesn't bind to the events?

Reproduction
------------

Install 7.15.5 - with starter kit - use a content item to create a Content Template - save (you should see the error) - go to the settings section - find the newly created content template - update one of the values - update will be saved but error message is displayed.

### Expected result

Content Template should be updated without throwing an error.

### Actual result

Content Template is saved but an error is thrown

---
_This item has been added to our backlog [AB#7858](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/7858)_

**Comment by jonat123** [comment]
Just tried to reproduce this issue on a fresh installation of Umbraco with a default starter kit and another clean installation with a few document types and I can confirm that I am seeing the same errors being thrown when trying to create and edit in the content templates.

**Comment by clausjensen** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/8675 - included in 7.15.6
